test/module/irohad/subscription/subscription_test.cpp

```diff
@@ -645,12 +662,65 @@ TEST_F(SubscriptionTest, Notify_2) {
   std::string test_data("test_data");
   uint32_t event_id = 10ul;
 
-  subscriber1->subscribe<1ull>(event_id);
-  subscriber2->subscribe<1ull>(event_id);
+  subscriber1->subscribe(1ull, event_id);
+  subscriber2->subscribe(1ull, event_id);
 
   EXPECT_CALL(*subscriber1, on_notify(0ull, event_id, std::string(test_data)))
       .Times(1);
   EXPECT_CALL(*subscriber2, on_notify(0ull, event_id, std::string(test_data)))
       .Times(1);
   engine->notify(event_id, test_data);
 }
+
+/**
+ * @given subscription engine and scheduler
+ * @when 2 subscribers are present
+ * @and they subscribe to different events in a single current scheduler
+ * @and both events are notified followed by dispose and process
+ * @then the handlers of each subscriber must be called once and process finish
+ * it's loop
+ */
+TEST_F(SubscriptionTest, InThreadDispatcherTest) {
+  auto manager = createSubscriptionManager<1>();
+  auto scheduler = std::make_shared<subscription::SchedulerBase>();
+
+  auto const current_thread_id = std::this_thread::get_id();
+  auto tid = manager->dispatcher()->bind(scheduler);
+  ASSERT_TRUE(tid);
+
+  utils::WaitForSingleObject complete[2];
```

Comment:
```suggestion
```
Can be removed because the test is single-threaded.

Reply:
fixed